torch/_dynamo/eval_frame.py

```diff
@@ -643,6 +645,177 @@ def __bool__(self):
         )
 
 
+class _AddRuntimeAssertsInInputConstraint(torch.fx.interpreter.Transformer):
+    def __init__(
+        self,
+        m,
+        constraints,
+        flat_args,
+    ):
+        super().__init__(m)
+        self.count = 0
+        self.flat_args = flat_args
+        self.constraints_id_to_constraint = defaultdict(list)
+        if constraints is not None:
+            # map input tensor to it's constraints
+            for constraint in constraints:
+                self.constraints_id_to_constraint[constraint.t_id].append(constraint)
+
+    def placeholder(self, target, args, kwargs):
+        arg = super().placeholder(target, args, kwargs)
+        orig_inp_id = id(self.flat_args[self.count])
+
+        if orig_inp_id not in self.constraints_id_to_constraint:
+            self.count += 1
+            return arg
+
+        constraints = self.constraints_id_to_constraint[orig_inp_id]
+
+        # Convert simple sympy Integers into concrete int to
+        # insert into graph
+        def _convert_to_int(val):
+            if val == sympy.oo:
+                return math.inf
+            if isinstance(val, sympy.Integer):
+                return int(val)
+            # TODO ignore expressions for now
+            return None
+
+        for constraint in constraints:
+            constraint_range = constraint.constraint_range
+            if constraint_range is None:
+                continue
+
+            min_int_val = _convert_to_int(constraint_range.vr.lower)
+            max_int_val = _convert_to_int(constraint_range.vr.upper)
+
+            if min_int_val is None and max_int_val is None:
+                continue
+
+            dim = self.tracer.create_proxy(
+                "call_function", torch.ops.aten.sym_size, (arg, constraint.dim), {}
+            )
+            assert_msg = f"Input #{self.count}'s dimension #{constraint.dim} size is outside of supported dynamic range"
```

Comment:
Nit, if you print orig_inp_id instead of self.count here, you can unconditionally increment self.count right after creating a new id without duplicating self.count = self.count + 1

Reply:
Hmm `orig_input_id` is actually a tensor python id tho (e.g id(tensor(5))).